app/src/main/java/com/dragic/gamehunter/view/uicomponents/ShimmerComponents.kt

```diff
@@ -0,0 +1,193 @@
+    modifier: Modifier,
+) {
+    if (isLoading) {
+        Column(
+            modifier = modifier
+                .height(dimensionResource(id = R.dimen.details_image_height))
+                .fillMaxWidth()
+                .shimmerEffect()
+        ) {}
+    } else {
+        contentAfterLoading()
+    }
+}
+
+@Composable
+fun ShimmerFeaturedDeals(
+    isLoading: Boolean,
+    contentAfterLoading: @Composable () -> Unit,
+    modifier: Modifier,
+) {
+    if (isLoading) {
+        Card(
+            modifier = modifier
+                .padding(
+                    horizontal = dimensionResource(id = R.dimen.details_deal_card_horizontal_padding),
+                    vertical = dimensionResource(id = R.dimen.details_deal_card_vertical_padding),
+                )
+                .height(dimensionResource(id = R.dimen.details_deal_card_height))
+                .fillMaxWidth()
+                .clip(RoundedCornerShape(dimensionResource(id = R.dimen.details_deal_card_radius)))
+        ) {
+            Row(
+                modifier = Modifier
+                    .fillMaxSize()
+            ) {
+                Box(
+                    modifier = Modifier
+                        .padding(dimensionResource(id = R.dimen.details_deal_card_store_image_padding))
+                        .size(dimensionResource(id = R.dimen.details_deal_card_store_image_size))
+                        .clip(CircleShape)
+                        .align(Alignment.CenterVertically)
+                        .shimmerEffect(),
+                )
+                Spacer(modifier = Modifier.size(dimensionResource(id = R.dimen.details_deal_card_horizontal_padding)))
+                Column(
+                    modifier = Modifier
+                        .fillMaxWidth()
+                        .padding(end = dimensionResource(id = R.dimen.deal_card_image_end_padding))
+                ) {
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .padding(top = dimensionResource(id = R.dimen.deal_card_title_top_padding))
+                            .height(dimensionResource(id = R.dimen.row_height))
+                            .shimmerEffect(),
+                    )
+                    Spacer(modifier = Modifier.size(dimensionResource(id = R.dimen.deal_card_text_padding)))
+                    Box(
+                        modifier = Modifier
+                            .fillMaxWidth()
+                            .height(dimensionResource(id = R.dimen.row_height))
+                            .shimmerEffect(),
+                    )
+                }
+            }
+        }
+
+    } else {
+        contentAfterLoading()
+    }
+
+}
+
+fun Modifier.shimmerEffect(): Modifier = composed {
+    val shimmerColors = listOf(
+        Color.LightGray.copy(alpha = 0.6f),
+        Color.LightGray.copy(alpha = 0.2f),
+        Color.LightGray.copy(alpha = 0.6f),
```

Comment:
This will construct a new `shimmerColors` on every call to the modifier. A better practice would be to extract it into a private val above the modifier definition: 

```kt
private val shimmerColors = listOf(...)
fun Modifier.shimmerEffect(): Modifier = composed {...}
```